tests/suites/test_suite_ssl.function

```diff
@@ -928,31 +928,86 @@ int mbedtls_move_handshake_to_state( mbedtls_ssl_context *ssl,
 #endif /* MBEDTLS_X509_CRT_PARSE_C */
 
 /*
- * Write application data. Then increase write and fragments counter
+ * Write application data. Increase write counter and fragments counter if
+ * necessary.
  */
 int mbedtls_ssl_write_fragment( mbedtls_ssl_context *ssl, unsigned char *buf,
-                                int ln, int *writen, int *fragments )
+                                int buf_len, int *written,
+                                int *fragments, const int expected_fragments )
 {
-    int ret = mbedtls_ssl_write( ssl, buf + *writen, ln - *writen );
-    if( ret >= 0 )
+    int ret = mbedtls_ssl_write( ssl, buf + *written, buf_len - *written );
+    if( ret > 0 )
     {
         (*fragments)++;
-        *writen += ret;
+        *written += ret;
     }
-    return ret;
+
+    if( expected_fragments == 0 )
+    {
+        /* Used for DTLS and the message size larger than MFL. This error must
+         * be returned for some of tests */
```

Comment:
Which tests must this be returned for? Update the C comment to be explicit about which tests.

"This error must be returned for tests that [insert criteria here]"

Reply:
This error is expected in case of negative test for message size larger than MFL and DTLS. It is used in mbedtls_exchange_data function to prevent dead loop which usually run until all data will be sent and received. MBEDTLS_ERR_SSL_BAD_INPUT_DATA seems to be general, but that error code is returned when there is too much data